### PR TITLE
fix: setrange, ltrim and lset command

### PR DESCRIPTION
**Pull request by Mario-LuX**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
This PR is intended to solve issue #24, #28 and #29 
Now these three commands behaves the same as original Redis

* **What is the new behavior (if this is a feature change)?**
None

### CODE REVIEW

File: src/t_pstring.c [hunk 1]
```diff
@@ -738,6 +738,7 @@ int pmSetCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 int setRangeFunc(const char *old_val, size_t old_val_len, char **new_val,
                  size_t *new_val_len, void *args_pointer) {
   // assert(args_pointer);
+  *new_val = (char *)malloc(MAX_KVDK_STRING_SIZE);
```
Comment (chencheng1560) [inline src/t_pstring.c:741]: malloc MAX_KVDK_STRING_SIZE at line 741 may trigger

memory leak at line 768
memory space waste if each time allocate a MAX_KVDK_STRING_SIZE space for a small string
if we just want to fix max length case for:
pm.setrange mykey 536870910 world

can we consider add something like if(offset+val_len > MAX_KVDK_STRING_SIZE) {return ERR; } around line 768?

File: src/t_plist.c [hunk 2]
```diff
@@ -338,11 +338,7 @@ int pmLsetCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   }
   size_t val_len;
   const char *val = RedisModule_StringPtrLen(argv[3], &val_len);
-  if (idx >= 0) {
-    KVDKListInsertAfter(engine, iter, val, val_len);
-  } else {
-    KVDKListInsertBefore(engine, iter, val, val_len);
-  }
+  s = KVDKListReplace(engine, iter, val, val_len);
```
Comment (chencheng1560) [inline src/t_plist.c:341]: can we confirm that the result of pm.lset is the same of the original behavior of redis after the modification?
E.g.
redis:6379> RPUSH mylist "one"
(integer) 1
redis:6379> RPUSH mylist "two"
(integer) 2
redis:6379> RPUSH mylist "three"
(integer) 3
redis:6379> LSET mylist 0 "four"
"OK"
redis:6379> LSET mylist -2 "five"
"OK"
redis:6379> LRANGE mylist 0 -1
1) "four"
2) "five"
3) "three"

File: src/t_plist.c [hunk 3]
```diff
@@ -420,6 +416,9 @@ int pmLtrimCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     if (start < 0) start = 0;
   }
   if (end < 0) end = list_sz + end;
+  if (end >= (long long)list_sz) end = list_sz - 1;
```
Comment (chencheng1560) [inline src/t_plist.c:419]: ok

**Review by chencheng1560 — CHANGES_REQUESTED** [review]
lset issue: pls confirm the pmedis' behavior is the same as the original redis.
setrange issue: pls confirm KVDK's new behavior

File: src/t_plist.c (hunk 2 above)
Comment (Mario-LuX) [reply src/t_plist.c:341]: Yes, this is confirmed.

Comment (chencheng1560) [reply src/t_plist.c:341]: Ok, good

**Review by chencheng1560 — APPROVED** [review]
ok, we can re-test ut after upgrade kvdk

**Pull request merged**